src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/SwaggerSamClient.scala

```diff
@@ -102,39 +106,77 @@ class SwaggerSamClient(samBasePath: String, cacheExpiryTime: FiniteDuration, cac
+    // this is called as the user's pet
+    val samAPI = samResourcesApiAsPet(userEmail, googleProject)
+    samAPI.createResource(notebookClusterResourceTypeName, getClusterResourceId(googleProject, clusterName))
   }
 
   def deleteNotebookClusterResource(creatorEmail: WorkbenchEmail, googleProject: GoogleProject, clusterName: ClusterName) = {
-    resourcesApiAsPet(creatorEmail, googleProject).deleteResource(notebookClusterResourceTypeName, getClusterResourceId(googleProject, clusterName))
+    logger.debug(s"Calling Sam deleteNotebookClusterResource as PET for user [${creatorEmail.value}] with project [${googleProject.value}] and clusterName [${clusterName.value}]")
+    // this is called as the user's pet
+    val samAPI = samResourcesApiAsPet(creatorEmail, googleProject)
+    samAPI.deleteResource(notebookClusterResourceTypeName, getClusterResourceId(googleProject, clusterName))
   }
 
-  def hasActionOnBillingProjectResource(userEmail: WorkbenchEmail, googleProject: GoogleProject, action: String): Boolean = {
-    hasActionOnResource(billingProjectResourceTypeName, googleProject.value, userEmail, googleProject, action)
+  def hasActionOnBillingProjectResource(userInfo: UserInfo, googleProject: GoogleProject, action: String): Boolean = {
+    // this is called with the user's token
+    hasActionOnResource(userInfo, billingProjectResourceTypeName, googleProject.value, action)
   }
 
-  def hasActionOnNotebookClusterResource(userEmail: WorkbenchEmail, googleProject: GoogleProject, clusterName: ClusterName, action: String): Boolean = {
-    hasActionOnResource(notebookClusterResourceTypeName, getClusterResourceId(googleProject, clusterName), userEmail, googleProject, action)
+  def hasActionOnNotebookClusterResource(userInfo: UserInfo, googleProject: GoogleProject, clusterName: ClusterName, action: String): Boolean = {
+    // this is called with the user's token
+    hasActionOnResource(userInfo, notebookClusterResourceTypeName, getClusterResourceId(googleProject, clusterName), action)
   }
 
   def getUserProxyFromSam(userEmail: WorkbenchEmail): WorkbenchEmail = {
+    logger.debug(s"Calling Sam getUserProxyFromSam as user ${leoEmail} with email ${userEmail.value}")
+    // this is called as the Leo service account
     val samAPI = samGoogleApi(getAccessTokenUsingPem(leoEmail, leoPem))
     WorkbenchEmail(samAPI.getProxyGroup(userEmail.value))
   }
 
-  def getPetServiceAccount(userEmail: WorkbenchEmail, googleProject: GoogleProject): WorkbenchEmail = {
-    WorkbenchEmail(googleApiAsPet(userEmail, googleProject).getPetServiceAccount(googleProject.value))
+  def getPetServiceAccount(userInfo: UserInfo, googleProject: GoogleProject): WorkbenchEmail = {
+    logger.debug(s"Calling Sam getPetServiceAccount as user [${userInfo}] and project [${googleProject.value}]")
+    // this is called with the user's token
+    val samAPI = samGoogleApi(userInfo.accessToken.token)
+    WorkbenchEmail(samAPI.getPetServiceAccount(googleProject.value))
```

Comment:
Workbench Model has `ServiceAccount` and `PetServiceAccount`. I think those are a bit overkill and unnecessary for this - but when I was debugging something earlier, I thought it would be cool if a user's email and their pet service account were distinct at a glance. 

I don't think this under the scope of this ticket though, just a thought :)

Reply:
That is a good thought. I don't think `ServiceAccount` and `PetServiceAccount` are quite right for this case because they contain things besides an email (e.g. an id, display name, etc) and in this case we just have an email. I wonder if we could fit `WorkbenchServiceAccountEmail` into the workbench-libs data model. I do think it's out of scope for this PR though. :)